frontend/test/pytest/test_conditionals.py

```diff
@@ -311,6 +311,29 @@ def cond_else():
 
         assert 0 == circuit()
 
+    def test_branch_multi_return_type_unification_qjit_2(self):
+        """Test that unification happens before the results of the cond primitve is available."""
+
+        @qjit
+        def circuit(cond1, cond2):
+            @cond(cond1)
+            def cond_fn():
+                return False
+
+            @cond_fn.else_if(cond2)
+            def cond_fn_2():
+                return 0.5
+
+            @cond_fn.otherwise
+            def cond_fn_3():
+                return False
+
+            r = cond_fn()
+            assert r.dtype is jnp.dtype("float64")
```

Comment:
It might be more future-proof to check if JAX 64-bit precision is enabled before asserting the type is `float64`, e.g.

```suggestion
            assert r.dtype is jnp.dtype("float64") if jax.config.values["jax_enable_x64"] else r.dtype is jnp.dtype("float32") 
```